lms/static/scripts/frontend_apps/utils/test/google-picker-client-test.js

```diff
@@ -106,35 +126,70 @@ describe('GooglePickerClient', () => {
     fakeGoogleLibs = createGoogleLibFakes();
     fakeLoadLibraries = sinon.stub().resolves(fakeGoogleLibs);
 
+    const fakeIdentityServicesLibrary = {
+      oauth2: {
+        initTokenClient: config => {
+          fakeTokenClient = new FakeTokenClient(config, authError);
+          return fakeTokenClient;
+        },
+      },
+    };
+
     $imports.$mock({
       './google-api-client': {
+        loadIdentityServicesLibrary: async () => fakeIdentityServicesLibrary,
         loadLibraries: fakeLoadLibraries,
       },
     });
   });
 
   afterEach(() => {
+    authError = null;
     $imports.$restore();
   });
 
   describe('#constructor', () => {
     it('loads Google API client', () => {
       createClient();
-      assert.calledWith(fakeLoadLibraries, ['auth2', 'client', 'picker']);
+      assert.calledWith(fakeLoadLibraries, ['client', 'picker']);
     });
   });
 
-  describe('#showPicker', () => {
+  describe('#requestAuthorization', () => {
     it("requests the user's authorization to access their Google Drive files", async () => {
       const client = createClient();
-      client.showPicker();
 
-      await fakeGoogleLibs.pickerVisible;
+      await client.requestAuthorization();
 
-      assert.calledWith(fakeGoogleLibs.auth2.init, {
+      assert.ok(fakeTokenClient);
+      assert.match(fakeTokenClient.config, {
         client_id: '12345',
         scope: GOOGLE_DRIVE_SCOPE,
       });
+      assert.calledOnce(fakeTokenClient.requestAccessToken);
+    });
+
+    it('does nothing if called a second time', async () => {
```

Comment:
Well, it returns a cached access token, right? I guess it doesn't do anything internally :)